features/consent/src/main/java/in/projecteka/jataayu/consent/ui/fragment/ConfirmPinFragment.kt

```diff
@@ -81,34 +81,40 @@ class ConfirmPinFragment : BaseDialogFragment(), OtpSubmissionClickHandler, OtpC
             bundle.getString(PIN)?.let { pin ->
                 if (confirmedPin == pin) {
                     showProgressBar(true)
-                    viewModel.createPinResponse.observe(this, Observer {
+                    viewModel.createPinResponse.observe(this, Observer { payloadResource ->
 
-                        when (it) {
-                            is Loading -> showProgressBar(it.isLoading, getString(R.string.creating_pin))
+                        when (payloadResource) {
+                            is Loading -> showProgressBar(payloadResource.isLoading, getString(R.string.creating_pin))
                             is Success -> {
-                                activity?.let {
-                                    it.setPinCreated(true)
+                                activity?.let { fragmentActivity ->
+                                    fragmentActivity.setPinCreated(true)
                                     showProgressBar(true)
-                                    viewModel.userVerificationResponse.observe(this, Observer { userVerificationResponse ->
-                                        activity?.setConsentTempToken(userVerificationResponse.temporaryToken)
-                                        EventBus.getDefault().post(MessageEventType.USER_VERIFIED)
-                                        it.setResult(Activity.RESULT_OK)
-                                        it.finish()
-                                    })
+                                    viewModel.userVerificationResponse.observe(
```

Comment:
should be outside this observer. Nesting observers leads to reregistering of observers everytime the parent observer observes an emission

Reply:
Good catch